lib/rubocop/cop/style/multiple_comparison.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks against comparing a variable with multiple items,
+      # where `Array.include?` could be used instead to avoid DRY errors.
+      #
+      # @example
+      #   # bad
+      #   if a == 'a' || a == 'b' || a == 'c'
+      #
+      #   # good
+      #   if ['a', 'b', 'c'].include? a
+      class MultipleComparison < Cop
+        MSG = 'Avoid comparing a variable with multiple items' \
+          'in a conditional, use include? instead.'.freeze
+
+        def_node_matcher :simple_comparison?, '(send $lvar :== _)'
```

Comment:
My understanding of `node_pattern.rb` was that `$lvar` will match against variable first, and then capture. It doesn't seem to match integers etc, but maybe there is other mechanism I don't fully understand?